### ISSUE TITLE
Cannot read property 'name' of undefined (Node, React, ES6, Babel project)

### ISSUE THREAD

**Issue by DWboutin** [jsdoc3/jsdoc#1163]
Hi, 

I'm trying to generate my doc but it crash very quickly. My project is built with Babel, Reactjs, Nodejs in ES6. I just uninstall jsdoc and re-installed it.

Why it produce this error? 

```sh
jsdoc ./src/**/** ./gulp/**/** -d ./docs
/usr/local/lib/node_modules/jsdoc/lib/jsdoc/src/astnode.js:214
                var key = prop.key.name;
                                  ^

TypeError: Cannot read property 'name' of undefined
    at /usr/local/lib/node_modules/jsdoc/lib/jsdoc/src/astnode.js:214:35
    at Array.forEach (native)
    at exports.nodeToValue (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/src/astnode.js:213:29)
    at Object.exports.getInfo (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/src/astnode.js:315:26)
    at Visitor.makeSymbolFoundEvent (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/src/visitor.js:502:33)
    at Visitor.visitNode (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/src/visitor.js:446:18)
    at Visitor.visit (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/src/visitor.js:368:27)
    at Walker.recurse (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/src/walker.js:591:44)
    at Parser._walkAst (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/src/parser.js:267:18)
    at Parser._parseSourceCode (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/src/parser.js:258:18)
    at Parser.parse (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/src/parser.js:179:18)
    at Object.module.exports.cli.parseFiles (/usr/local/lib/node_modules/jsdoc/cli.js:356:42)
    at module.exports.cli.main (/usr/local/lib/node_modules/jsdoc/cli.js:226:14)
    at Object.module.exports.cli.runCommand (/usr/local/lib/node_modules/jsdoc/cli.js:180:5)
    at /usr/local/lib/node_modules/jsdoc/jsdoc.js:103:9
    at Object.<anonymous> (/usr/local/lib/node_modules/jsdoc/jsdoc.js:104:3)
```

**Comment by DWboutin** [comment]
Anybody have an idea for this error? I try to create the docs for this file and it failed the same way:

``javascript
/**
 * @file Conteneur Formulaire d'inscription - Signup Market informations
 * @author Mikael Boutin
 * @version 0.0.1
 */
import React from 'react';
import { browserHistory } from 'react-router';

import CampaignMarket from './CampaignMarket.react.js';

import Animations from '../../utils/Animations';

const formName = 'market';

/**
 * Conteneur du formulaire d'inscription Contact infos - Sert à gérer les transition avec ReactTransitionGroup
 */
class CampaignMarketContainer extends React.Component {

  /**
   * @constructor
   * @param {object} props - La fonction super() appelle le parent pour y transmettre ses propriétés
   * @param {object} this.animations - Initialise la classe d'animation avec les classes CSS pour ReactTransitionGroup https://facebook.github.io/react/docs/animation.html#low-level-api-reacttransitiongroup
   */
  constructor(props) {
    super(props);
    this.animations = new Animations('#campaign-market');

    this.handleFormSubmit = this.handleFormSubmit.bind(this);
  }

  /**
   * Reçoit les valeurs des formulaires
   */
  handleFormSubmit(data) {
    const { submitForm } = this.props;

    const currentForm = {};
    currentForm[formName] = {
      ...data,
      completed: true,
    };

    submitForm(currentForm);

    setTimeout(() => {
      browserHistory.push('/signup/references');
    }, 50);
  }

  /**
   * Première apparition du component - Gère l'animation
   * @param {function} callback - La fonction callback doit être appelée pour procéder aux prochaines étapes
   */
  componentWillAppear(callback) {
    this.animations.animationClass('registration-sl-appear', 500, callback);
  }

  /**
   * Entrée du component dans le DOM - Gère l'animation
   * @param {function} callback - La fonction callback doit être appelée pour procéder aux prochaines étapes
   */
  componentWillEnter(callback) {
    this.animations.animationClass('registration-sl-enter', 500, callback);
  }

  /**
   * Lorsque le component quitte - Gère l'animation
   * @param {function} callback - La fonction sert à enlever le component du DOM
   */
  componentWillLeave(callback) {
    this.animations.animationClass('registration-sl-leave', 500, callback);
  }

  /**
   * Render le component RegistrationConnectionInfos
   * @return {JSX} - Rend la partie du formulaire d'inscription Connection infos
   */
  render() {
    return (
      <CampaignMarket {...this.props} onSubmit={this.handleFormSubmit} />
    );
  }
}

/**
 * Propriétés du component
 * @type {object}
 * @property {function} submitForm - Fonction de soumission de formulaire.
 */
CampaignMarketContainer.propTypes = {
  submitForm: React.PropTypes.func,
};

export default CampaignMarketContainer;
```

**Comment by brabeji** [comment]
I get the same error with version `3.4.0`. Spread operator is suspect here as the minimal failing example is:
```
var a = {a: 'a'};
var b = {b: 'b'};
var result = {...a, ...b};
```

**Comment by brabeji** [comment]
`astnode.js` has no support for node of type `ExperimentalSpreadProperty` which seems reasonable. Not sure what the best next step is.

**Comment by DWboutin** [comment]
Any progress on this?

**Comment by rhys-vdw** [comment]
Looks like [this might be fixed in master](https://github.com/jsdoc3/jsdoc/blob/407bd47/lib/jsdoc/src/astnode.js#L226-L230).

**Comment by NervosaX** [comment]
A workaround for now is using a babel plugin:

```
npm install --save-dev jsdoc-babel babel-plugin-transform-object-rest-spread
```

```json
{
	"opts": {
		"template": "node_modules/docdash",
		"destination": "./docs",
		"recurse": true,
		"readme": "README.md"
	},
	"plugins": ["node_modules/jsdoc-babel"],
	"babel": {
		"plugins": ["transform-object-rest-spread"]
	}
}
```

**Comment by sashafklein** [comment]
@rhys-vdw seems to be right that the bug has been fixed in master. 

We changed our `package.json` file to point to the newest commit (as opposed to an official release):

```
    "jsdoc": "git://github.com/jsdoc3/jsdoc.git#407bd470cde2dcb2745c217411fe8f9406cb694d",
```

That fixed the problem, without the need for any additional babel plugins or node modules.

**Comment by hegemonic** [comment]
As noted, this issue is now fixed on master. It's also fixed in version 3.4.1, which was just released.
